"use strict";(self.webpackChunkGATK_SV=self.webpackChunkGATK_SV||[]).push([[3976],{1512:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>u,frontMatter:()=>r,metadata:()=>a,toc:()=>c});var i=n(4848),s=n(8453);const r={title:"About",description:"GATK-SV",sidebar_position:1},o=void 0,a={id:"intro",title:"About",description:"GATK-SV",source:"@site/docs/intro.md",sourceDirName:".",slug:"/intro",permalink:"/gatk-sv/docs/intro",draft:!1,unlisted:!1,editUrl:"https://github.com/broadinstitute/gatk-sv/tree/master/website/docs/intro.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{title:"About",description:"GATK-SV",sidebar_position:1},sidebar:"tutorialSidebar",next:{title:"Getting Started",permalink:"/gatk-sv/docs/category/getting-started"}},l={},c=[{value:"Citation",id:"citation",level:3}];function d(e){const t={a:"a",h3:"h3",li:"li",p:"p",ul:"ul",...(0,s.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsxs)(t.p,{children:["GATK-SV is a comprehensive, cloud-based ensemble pipeline to capture and annotate all\nclasses of structural variants (SV) from whole genome sequencing (WGS). It can detect\ndeletions, duplications, multi-allelic copy number variants, balanced inversions,\ninsertions, translocations, and a diverse spectrum of complex SV. Briefly, GATK-SV\nmaximizes the sensitivity of SV discovery by harmonizing output from five tools\n(Manta, Wham, cnMOPS, GATK-gCNV, MELT). In order to reduce false positives, raw SV\nare adjudicated and re-genotyped from read evidence considering all potential\nsequencing evidence including anomalous paired-end (PE) reads, split reads (SR) crossing\na breakpoint, normalized read-depth (RD), and B-allele frequencies (BAF). It also\nfully resolves 11 classes of complex SV based on their breakpoint signature. GATK-SV\nhas been designed to be deployed in the Google Cloud Platform via the cromwell\nexecution engine, which allows massively parallel scaling. Further details about\nGATK--SV can be found in ",(0,i.jsx)(t.a,{href:"https://www.nature.com/articles/s41586-020-2287-8",children:"Collins et al. 2020, Nature"}),"."]}),"\n",(0,i.jsxs)(t.p,{children:["A high-level description of GATK-SV is available ",(0,i.jsx)(t.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/9022487952155-Structural-variant-SV-discovery",children:"here"}),"."]}),"\n",(0,i.jsx)(t.h3,{id:"citation",children:"Citation"}),"\n",(0,i.jsx)(t.p,{children:"Please cite the following publication:"}),"\n",(0,i.jsxs)(t.ul,{children:["\n",(0,i.jsx)(t.li,{children:(0,i.jsx)(t.a,{href:"https://doi.org/10.1038/s41586-020-2287-8",children:'Collins, Brand, et al. 2020. "A structural variation reference for medical and population genetics." Nature 581, 444-451.'})}),"\n"]}),"\n",(0,i.jsx)(t.p,{children:"Additional references:"}),"\n",(0,i.jsxs)(t.ul,{children:["\n",(0,i.jsx)(t.li,{children:(0,i.jsx)(t.a,{href:"https://doi.org/10.1038/s41588-018-0107-y",children:'Werling et al. 2018. "An analytical framework for whole-genome sequence association studies and its implications for autism spectrum disorder." Nature genetics 50.5, 727-736.'})}),"\n"]})]})}function u(e={}){const{wrapper:t}={...(0,s.R)(),...e.components};return t?(0,i.jsx)(t,{...e,children:(0,i.jsx)(d,{...e})}):d(e)}},8453:(e,t,n)=>{n.d(t,{R:()=>o,x:()=>a});var i=n(6540);const s={},r=i.createContext(s);function o(e){const t=i.useContext(r);return i.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function a(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(s):e.components||s:o(e.components),i.createElement(r.Provider,{value:t},e.children)}}}]);