"use strict";(self.webpackChunkGATK_SV=self.webpackChunkGATK_SV||[]).push([[3309],{460:(e,s,n)=>{n.r(s),n.d(s,{assets:()=>d,contentTitle:()=>r,default:()=>h,frontMatter:()=>c,metadata:()=>t,toc:()=>o});var i=n(4848),l=n(8453);const c={title:"GatherBatchEvidence",description:"Gather Batch Evidence",sidebar_position:4,slug:"gbe"},r=void 0,t={id:"modules/gather_batch_evidence",title:"GatherBatchEvidence",description:"Gather Batch Evidence",source:"@site/docs/modules/gather_batch_evidence.md",sourceDirName:"modules",slug:"/modules/gbe",permalink:"/gatk-sv/docs/modules/gbe",draft:!1,unlisted:!1,editUrl:"https://github.com/broadinstitute/gatk-sv/tree/master/website/docs/modules/gather_batch_evidence.md",tags:[],version:"current",sidebarPosition:4,frontMatter:{title:"GatherBatchEvidence",description:"Gather Batch Evidence",sidebar_position:4,slug:"gbe"},sidebar:"tutorialSidebar",previous:{title:"TrainGCNV",permalink:"/gatk-sv/docs/modules/gcnv"},next:{title:"ClusterBatch",permalink:"/gatk-sv/docs/modules/cb"}},d={},o=[{value:"Inputs",id:"inputs",level:2},{value:"<code>batch</code>",id:"batch",level:4},{value:"<code>samples</code>",id:"samples",level:4},{value:"<code>counts</code>",id:"counts",level:4},{value:"Raw calls",id:"raw-calls",level:4},{value:"<code>PE_files</code>",id:"pe_files",level:4},{value:"<code>SR_files</code>",id:"sr_files",level:4},{value:"<code>SD_files</code>",id:"sd_files",level:4},{value:"<code>matrix_qc_distance</code>",id:"matrix_qc_distance",level:4},{value:"<code>min_svsize</code>",id:"min_svsize",level:4},{value:"<code>ped_file</code>",id:"ped_file",level:4},{value:"<code>run_matrix_qc</code>",id:"run_matrix_qc",level:4},{value:"<code>gcnv_qs_cutoff</code>",id:"gcnv_qs_cutoff",level:4},{value:"cn.MOPS files",id:"cnmops-files",level:4},{value:"GATK-gCNV inputs",id:"gatk-gcnv-inputs",level:4},{value:"Docker images",id:"docker-images",level:4},{value:"Static inputs",id:"static-inputs",level:4},{value:"Optional Inputs",id:"optional-inputs",level:4},{value:"Outputs",id:"outputs",level:2}];function a(e){const s={a:"a",code:"code",h2:"h2",h4:"h4",li:"li",mermaid:"mermaid",p:"p",pre:"pre",ul:"ul",...(0,l.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsxs)(s.p,{children:["Runs CNV callers (",(0,i.jsx)(s.a,{href:"https://academic.oup.com/nar/article/40/9/e69/1136601",children:"cn.MOPS"}),", GATK-gCNV)\nand combines single-sample raw evidence into a batch."]}),"\n",(0,i.jsxs)(s.p,{children:["The following diagram illustrates the downstream workflows of the ",(0,i.jsx)(s.code,{children:"GatherBatchEvidence"})," workflow\nin the recommended invocation order. You may refer to\n",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/terra_pipeline_diagram.jpg",children:"this diagram"}),"\nfor the overall recommended invocation order."]}),"\n",(0,i.jsx)(s.mermaid,{value:"\nstateDiagram\n  direction LR\n  \n  classDef inModules stroke-width:0px,fill:#00509d,color:#caf0f8\n  classDef thisModule font-weight:bold,stroke-width:0px,fill:#ff9900,color:white\n  classDef outModules stroke-width:0px,fill:#caf0f8,color:#00509d\n\n  gbe: GatherBatchEvidence\n  t: TrainGCNV\n  cb: ClusterBatch\n  t --\x3e gbe\n  gbe --\x3e cb\n  \n  class gbe thisModule\n  class t inModules\n  class cb outModules"}),"\n",(0,i.jsx)(s.h2,{id:"inputs",children:"Inputs"}),"\n",(0,i.jsx)(s.p,{children:"This workflow takes as input the read counts, BAF, PE, SD, SR, and per-caller VCF files\nproduced in the GatherSampleEvidence workflow, and contig ploidy and gCNV models from\nthe TrainGCNV workflow.\nThe following is the list of the inputs the GatherBatchEvidence workflow takes."}),"\n",(0,i.jsx)(s.h4,{id:"batch",children:(0,i.jsx)(s.code,{children:"batch"})}),"\n",(0,i.jsx)(s.p,{children:"An identifier for the batch."}),"\n",(0,i.jsx)(s.h4,{id:"samples",children:(0,i.jsx)(s.code,{children:"samples"})}),"\n",(0,i.jsx)(s.p,{children:"Sets the list of sample IDs."}),"\n",(0,i.jsx)(s.h4,{id:"counts",children:(0,i.jsx)(s.code,{children:"counts"})}),"\n",(0,i.jsxs)(s.p,{children:["Set to the ",(0,i.jsx)(s.a,{href:"./gse#coverage-counts",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.coverage_counts"})})," output."]}),"\n",(0,i.jsx)(s.h4,{id:"raw-calls",children:"Raw calls"}),"\n",(0,i.jsxs)(s.p,{children:["The following inputs set the per-caller raw SV calls, and should be set\nif the caller was run in the ",(0,i.jsx)(s.a,{href:"./gse",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence"})})," workflow.\nYou may set each of the following inputs to the linked output from\nthe GatherSampleEvidence workflow."]}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"manta_vcfs"}),": ",(0,i.jsx)(s.a,{href:"./gse#manta-vcf",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.manta_vcf"})}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"melt_vcfs"}),": ",(0,i.jsx)(s.a,{href:"./gse#melt-vcf",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.melt_vcf"})}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"scramble_vcfs"}),": ",(0,i.jsx)(s.a,{href:"./gse#scramble-vcf",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.scramble_vcf"})}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"wham_vcfs"}),": ",(0,i.jsx)(s.a,{href:"./gse#wham-vcf",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.wham_vcf"})}),"."]}),"\n"]}),"\n",(0,i.jsx)(s.h4,{id:"pe_files",children:(0,i.jsx)(s.code,{children:"PE_files"})}),"\n",(0,i.jsxs)(s.p,{children:["Set to the ",(0,i.jsx)(s.a,{href:"./gse#pesr-disc",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.pesr_disc"})})," output."]}),"\n",(0,i.jsx)(s.h4,{id:"sr_files",children:(0,i.jsx)(s.code,{children:"SR_files"})}),"\n",(0,i.jsxs)(s.p,{children:["Set to the ",(0,i.jsx)(s.a,{href:"./gse#pesr-split",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.pesr_split"})})]}),"\n",(0,i.jsx)(s.h4,{id:"sd_files",children:(0,i.jsx)(s.code,{children:"SD_files"})}),"\n",(0,i.jsxs)(s.p,{children:["Set to the ",(0,i.jsx)(s.a,{href:"./gse#pesr-sd",children:(0,i.jsx)(s.code,{children:"GatherSampleEvidence.pesr_sd"})})]}),"\n",(0,i.jsx)(s.h4,{id:"matrix_qc_distance",children:(0,i.jsx)(s.code,{children:"matrix_qc_distance"})}),"\n",(0,i.jsxs)(s.p,{children:["You may refer to ",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/GatherBatchEvidence.json.tmpl",children:"this file"}),"\nfor an example value."]}),"\n",(0,i.jsx)(s.h4,{id:"min_svsize",children:(0,i.jsx)(s.code,{children:"min_svsize"})}),"\n",(0,i.jsx)(s.p,{children:"Sets the minimum size of SVs to include."}),"\n",(0,i.jsx)(s.h4,{id:"ped_file",children:(0,i.jsx)(s.code,{children:"ped_file"})}),"\n",(0,i.jsxs)(s.p,{children:["A pedigree file describing the familial relationshipts between the samples in the cohort.\nPlease refer to ",(0,i.jsx)(s.a,{href:"./#ped_file",children:"this section"})," for details."]}),"\n",(0,i.jsx)(s.h4,{id:"run_matrix_qc",children:(0,i.jsx)(s.code,{children:"run_matrix_qc"})}),"\n",(0,i.jsx)(s.p,{children:"Enables or disables running optional QC tasks."}),"\n",(0,i.jsx)(s.h4,{id:"gcnv_qs_cutoff",children:(0,i.jsx)(s.code,{children:"gcnv_qs_cutoff"})}),"\n",(0,i.jsxs)(s.p,{children:["You may refer to ",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/GatherBatchEvidence.json.tmpl",children:"this file"}),"\nfor an example value."]}),"\n",(0,i.jsx)(s.h4,{id:"cnmops-files",children:"cn.MOPS files"}),"\n",(0,i.jsx)(s.p,{children:"The workflow needs the following cn.MOPS files."}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsxs)(s.li,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"cnmops_chrom_file"})," and ",(0,i.jsx)(s.code,{children:"cnmops_allo_file"}),": FASTA index files (",(0,i.jsx)(s.code,{children:".fai"}),") for respectively\nnon-sex chromosomes (autosomes) and chromosomes X and Y (allosomes).\nThe file format is explained ",(0,i.jsx)(s.a,{href:"https://www.htslib.org/doc/faidx.html",children:"on this page"}),"."]}),"\n",(0,i.jsx)(s.p,{children:"You may use the following files for these fields:"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-json",children:'"cnmops_chrom_file": "gs://gcp-public-data--broad-references/hg38/v0/sv-resources/resources/v1/autosome.fai"\n"cnmops_allo_file": "gs://gcp-public-data--broad-references/hg38/v0/sv-resources/resources/v1/allosome.fai"\n'})}),"\n"]}),"\n",(0,i.jsxs)(s.li,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"cnmops_exclude_list"}),":\nYou may use ",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/d66f760865a89f30dbce456a3f720dec8b70705c/inputs/values/resources_hg38.json#L10",children:"this file"}),"\nfor this field."]}),"\n"]}),"\n"]}),"\n",(0,i.jsx)(s.h4,{id:"gatk-gcnv-inputs",children:"GATK-gCNV inputs"}),"\n",(0,i.jsxs)(s.p,{children:["The following inputs are configured based on the outputs generated in the ",(0,i.jsx)(s.a,{href:"./gcnv",children:(0,i.jsx)(s.code,{children:"TrainGCNV"})})," workflow."]}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"contig_ploidy_model_tar"}),": ",(0,i.jsx)(s.a,{href:"./gcnv#contig-ploidy-model-tarball",children:(0,i.jsx)(s.code,{children:"TrainGCNV.cohort_contig_ploidy_model_tar"})})]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"gcnv_model_tars"}),": ",(0,i.jsx)(s.a,{href:"./gcnv#model-tarballs",children:(0,i.jsx)(s.code,{children:"TrainGCNV.cohort_gcnv_model_tars"})})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["The workflow also enables setting a few optional arguments of gCNV.\nThe arguments and their default values are provided\n",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/GatherBatchEvidence.json.tmpl",children:"here"}),"\nas the following, and each argument is documented on\n",(0,i.jsx)(s.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360037593411-PostprocessGermlineCNVCalls",children:"this page"}),"\nand\n",(0,i.jsx)(s.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360047217671-GermlineCNVCaller",children:"this page"}),"."]}),"\n",(0,i.jsx)(s.h4,{id:"docker-images",children:"Docker images"}),"\n",(0,i.jsxs)(s.p,{children:["The workflow needs the following Docker images, the latest versions of which are in\n",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/values/dockers.json",children:"this file"}),"."]}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"cnmops_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"condense_counts_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"linux_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"sv_base_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"sv_base_mini_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"sv_pipeline_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"sv_pipeline_qc_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"gcnv_gatk_docker"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"gatk_docker"}),"."]}),"\n"]}),"\n",(0,i.jsx)(s.h4,{id:"static-inputs",children:"Static inputs"}),"\n",(0,i.jsxs)(s.p,{children:["You may refer to ",(0,i.jsx)(s.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/values/resources_hg38.json",children:"this reference file"}),"\nfor values of the following inputs."]}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"primary_contigs_fai"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"cytoband"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"ref_dict"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"mei_bed"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"genome_file"}),";"]}),"\n",(0,i.jsxs)(s.li,{children:[(0,i.jsx)(s.code,{children:"sd_locs_vcf"}),"."]}),"\n"]}),"\n",(0,i.jsx)(s.h4,{id:"optional-inputs",children:"Optional Inputs"}),"\n",(0,i.jsx)(s.p,{children:"The following is the list of a few optional inputs of the\nworkflow, with an example of possible values."}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsx)(s.li,{children:(0,i.jsx)(s.code,{children:'"allosomal_contigs": [["chrX", "chrY"]]'})}),"\n",(0,i.jsx)(s.li,{children:(0,i.jsx)(s.code,{children:'"ploidy_sample_psi_scale": 0.001'})}),"\n"]}),"\n",(0,i.jsx)(s.h2,{id:"outputs",children:"Outputs"}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsx)(s.li,{children:"Combined read count matrix, SR, PE, and BAF files"}),"\n",(0,i.jsx)(s.li,{children:"Standardized call VCFs"}),"\n",(0,i.jsx)(s.li,{children:"Depth-only (DEL/DUP) calls"}),"\n",(0,i.jsx)(s.li,{children:"Per-sample median coverage estimates"}),"\n",(0,i.jsx)(s.li,{children:"(Optional) Evidence QC plots"}),"\n"]})]})}function h(e={}){const{wrapper:s}={...(0,l.R)(),...e.components};return s?(0,i.jsx)(s,{...e,children:(0,i.jsx)(a,{...e})}):a(e)}},8453:(e,s,n)=>{n.d(s,{R:()=>r,x:()=>t});var i=n(6540);const l={},c=i.createContext(l);function r(e){const s=i.useContext(c);return i.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function t(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(l):e.components||l:r(e.components),i.createElement(c.Provider,{value:s},e.children)}}}]);