"use strict";(self.webpackChunkGATK_SV=self.webpackChunkGATK_SV||[]).push([[6030],{4588:(e,n,i)=>{i.r(n),i.d(n,{assets:()=>c,contentTitle:()=>l,default:()=>h,frontMatter:()=>r,metadata:()=>o,toc:()=>d});var s=i(4848),t=i(8453);const r={title:"Overview",description:"Overview of the constituting components",sidebar_position:0},l=void 0,o={id:"modules/index",title:"Overview",description:"Overview of the constituting components",source:"@site/docs/modules/index.md",sourceDirName:"modules",slug:"/modules/",permalink:"/gatk-sv/docs/modules/",draft:!1,unlisted:!1,editUrl:"https://github.com/broadinstitute/gatk-sv/tree/master/website/docs/modules/index.md",tags:[],version:"current",sidebarPosition:0,frontMatter:{title:"Overview",description:"Overview of the constituting components",sidebar_position:0},sidebar:"tutorialSidebar",previous:{title:"Modules",permalink:"/gatk-sv/docs/category/modules"},next:{title:"GatherSampleEvidence",permalink:"/gatk-sv/docs/modules/gse"}},c={},d=[{value:"Pipeline Parameters",id:"pipeline-parameters",level:2},{value:"<code>ped_file</code>",id:"ped_file",level:4}];function a(e){const n={a:"a",code:"code",h2:"h2",h4:"h4",li:"li",p:"p",strong:"strong",ul:"ul",...(0,t.R)(),...e.components};return(0,s.jsxs)(s.Fragment,{children:[(0,s.jsxs)(n.p,{children:["The pipeline is written in ",(0,s.jsx)(n.a,{href:"https://openwdl.org",children:"Workflow Description Language (WDL)"}),",\nconsisting of multiple modules to be executed in the following order."]}),"\n",(0,s.jsxs)(n.ul,{children:["\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"GatherSampleEvidence"})," SV evidence collection, including calls from a configurable set of\nalgorithms (Manta, MELT, and Wham), read depth (RD), split read positions (SR),\nand discordant pair positions (PE)."]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"EvidenceQC"})," Dosage bias scoring and ploidy estimation."]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"GatherBatchEvidence"})," Copy number variant calling using\n",(0,s.jsx)(n.code,{children:"cn.MOPS"})," and ",(0,s.jsx)(n.code,{children:"GATK gCNV"}),"; B-allele frequency (BAF) generation;\ncall and evidence aggregation."]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"ClusterBatch"})," Variant clustering"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"GenerateBatchMetrics"})," Variant filtering metric generation"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"FilterBatch"})," Variant filtering; outlier exclusion"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"GenotypeBatch"})," Genotyping"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"MakeCohortVcf"})," Cross-batch integration; complex variant resolution and re-genotyping; vcf cleanup"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"Module 07 (in development)"})," Downstream filtering, including minGQ, batch effect check,\noutlier samples removal and final recalibration;"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"AnnotateVCF"})," Annotations, including functional annotation,\nallele frequency (AF) annotation and AF annotation with external population callsets;"]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsxs)(n.p,{children:[(0,s.jsx)(n.strong,{children:"Module 09 (in development)"})," Visualization, including scripts that generates IGV screenshots and rd plots."]}),"\n"]}),"\n",(0,s.jsxs)(n.li,{children:["\n",(0,s.jsx)(n.p,{children:"Additional modules to be added: de novo and mosaic scripts"}),"\n"]}),"\n"]}),"\n",(0,s.jsx)(n.h2,{id:"pipeline-parameters",children:"Pipeline Parameters"}),"\n",(0,s.jsx)(n.p,{children:"Several inputs are shared across different modules of the pipeline, which are explained in this section."}),"\n",(0,s.jsx)(n.h4,{id:"ped_file",children:(0,s.jsx)(n.code,{children:"ped_file"})}),"\n",(0,s.jsxs)(n.p,{children:["A pedigree file describing the familial relationships between the samples in the cohort.\nThe file needs to be in the\n",(0,s.jsx)(n.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035531972-PED-Pedigree-format",children:"PED format"}),".\nUpdated with ",(0,s.jsx)(n.a,{href:"./eqc",children:"EvidenceQC"})," sex assignments, including\n",(0,s.jsx)(n.code,{children:"sex = 0"})," for sex aneuploidies;\ngenotypes on chrX and chrY for samples with ",(0,s.jsx)(n.code,{children:"sex = 0"})," in the PED file will be set to\n",(0,s.jsx)(n.code,{children:"./."})," and these samples will be excluded from sex-specific training steps."]})]})}function h(e={}){const{wrapper:n}={...(0,t.R)(),...e.components};return n?(0,s.jsx)(n,{...e,children:(0,s.jsx)(a,{...e})}):a(e)}},8453:(e,n,i)=>{i.d(n,{R:()=>l,x:()=>o});var s=i(6540);const t={},r=s.createContext(t);function l(e){const n=s.useContext(r);return s.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function o(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(t):e.components||t:l(e.components),s.createElement(r.Provider,{value:n},e.children)}}}]);