"use strict";(self.webpackChunkGATK_SV=self.webpackChunkGATK_SV||[]).push([[6367],{2366:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>c,contentTitle:()=>r,default:()=>u,frontMatter:()=>o,metadata:()=>l,toc:()=>a});var s=n(4848),i=n(8453);const o={title:"MakeCohortVcf",description:"Make Cohort VCF",sidebar_position:11,slug:"cvcf"},r=void 0,l={id:"modules/make_cohort_vcf",title:"MakeCohortVcf",description:"Make Cohort VCF",source:"@site/docs/modules/make_cohort_vcf.md",sourceDirName:"modules",slug:"/modules/cvcf",permalink:"/gatk-sv/docs/modules/cvcf",draft:!1,unlisted:!1,editUrl:"https://github.com/broadinstitute/gatk-sv/tree/master/website/docs/modules/make_cohort_vcf.md",tags:[],version:"current",sidebarPosition:11,frontMatter:{title:"MakeCohortVcf",description:"Make Cohort VCF",sidebar_position:11,slug:"cvcf"},sidebar:"tutorialSidebar",previous:{title:"ReGenotypeCNVs",permalink:"/gatk-sv/docs/modules/rgcnvs"},next:{title:"Downstream Filtering",permalink:"/gatk-sv/docs/modules/df"}},c={},a=[{value:"Prerequisites",id:"prerequisites",level:3},{value:"Inputs",id:"inputs",level:3},{value:"Outputs",id:"outputs",level:3}];function d(e){const t={h3:"h3",li:"li",p:"p",ul:"ul",...(0,i.R)(),...e.components};return(0,s.jsxs)(s.Fragment,{children:[(0,s.jsx)(t.p,{children:"Combines variants across multiple batches, resolves complex variants,\nre-genotypes, and performs final VCF clean-up."}),"\n",(0,s.jsx)(t.h3,{id:"prerequisites",children:"Prerequisites"}),"\n",(0,s.jsxs)(t.ul,{children:["\n",(0,s.jsx)(t.li,{children:"GenotypeBatch"}),"\n",(0,s.jsx)(t.li,{children:"(Optional) RegenotypeCNVs"}),"\n"]}),"\n",(0,s.jsx)(t.h3,{id:"inputs",children:"Inputs"}),"\n",(0,s.jsxs)(t.ul,{children:["\n",(0,s.jsx)(t.li,{children:"RD, PE and SR file URIs (GatherBatchEvidence)"}),"\n",(0,s.jsx)(t.li,{children:"Batch filtered PED file URIs (FilterBatch)"}),"\n",(0,s.jsx)(t.li,{children:"Genotyped PESR VCF URIs (GenotypeBatch)"}),"\n",(0,s.jsx)(t.li,{children:"Genotyped depth VCF URIs (GenotypeBatch or RegenotypeCNVs)"}),"\n",(0,s.jsx)(t.li,{children:"SR pass variant file URIs (GenotypeBatch)"}),"\n",(0,s.jsx)(t.li,{children:"SR fail variant file URIs (GenotypeBatch)"}),"\n",(0,s.jsx)(t.li,{children:"Genotyping cutoff file URIs (GenotypeBatch)"}),"\n",(0,s.jsx)(t.li,{children:"Batch IDs"}),"\n",(0,s.jsx)(t.li,{children:"Sample ID list URIs"}),"\n"]}),"\n",(0,s.jsx)(t.h3,{id:"outputs",children:"Outputs"}),"\n",(0,s.jsxs)(t.ul,{children:["\n",(0,s.jsx)(t.li,{children:'Finalized "cleaned" VCF and QC plots'}),"\n"]})]})}function u(e={}){const{wrapper:t}={...(0,i.R)(),...e.components};return t?(0,s.jsx)(t,{...e,children:(0,s.jsx)(d,{...e})}):d(e)}},8453:(e,t,n)=>{n.d(t,{R:()=>r,x:()=>l});var s=n(6540);const i={},o=s.createContext(i);function r(e){const t=s.useContext(o);return s.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function l(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(i):e.components||i:r(e.components),s.createElement(o.Provider,{value:t},e.children)}}}]);