"use strict";(self.webpackChunkGATK_SV=self.webpackChunkGATK_SV||[]).push([[6065],{9255:(e,t,s)=>{s.r(t),s.d(t,{assets:()=>a,contentTitle:()=>o,default:()=>u,frontMatter:()=>i,metadata:()=>l,toc:()=>c});var n=s(4848),r=s(8453);const i={title:"Downstream Filtering",description:"Downstream filtering (work in progress)",sidebar_position:12,slug:"df"},o=void 0,l={id:"modules/downstream_filter",title:"Downstream Filtering",description:"Downstream filtering (work in progress)",source:"@site/docs/modules/downstream_filter.md",sourceDirName:"modules",slug:"/modules/df",permalink:"/gatk-sv/docs/modules/df",draft:!1,unlisted:!1,editUrl:"https://github.com/broadinstitute/gatk-sv/tree/master/website/docs/modules/downstream_filter.md",tags:[],version:"current",sidebarPosition:12,frontMatter:{title:"Downstream Filtering",description:"Downstream filtering (work in progress)",sidebar_position:12,slug:"df"},sidebar:"tutorialSidebar",previous:{title:"MakeCohortVcf",permalink:"/gatk-sv/docs/modules/cvcf"},next:{title:"AnnotateVCF",permalink:"/gatk-sv/docs/modules/av"}},a={},c=[];function d(e){const t={code:"code",li:"li",p:"p",pre:"pre",ul:"ul",...(0,r.R)(),...e.components};return(0,n.jsxs)(n.Fragment,{children:[(0,n.jsx)(t.p,{children:"Apply downstream filtering steps to the cleaned VCF to further\ncontrol the false discovery rate; all steps are optional\nand users should decide based on the specific purpose of\ntheir projects."}),"\n",(0,n.jsx)(t.p,{children:"Filtering methods include:"}),"\n",(0,n.jsxs)(t.ul,{children:["\n",(0,n.jsxs)(t.li,{children:["\n",(0,n.jsx)(t.p,{children:"minGQ - remove variants based on the genotype quality across\npopulations. Note: Trio families are required to build the minGQ\nfiltering model in this step. We provide tables pre-trained with\nthe 1000 genomes samples at different FDR thresholds for projects\nthat lack family structures, and they can be found at the paths below.\nThese tables assume that GQ has a scale of [0,999], so they will\nnot work with newer VCFs where GQ has a scale of [0,99]."}),"\n",(0,n.jsx)(t.pre,{children:(0,n.jsx)(t.code,{className:"language-shell",children:"gs://gatk-sv-resources-public/hg38/v0/sv-resources/ref-panel/1KG/v2/mingq/1KGP_2504_and_698_with_GIAB.10perc_fdr.PCRMINUS.minGQ.filter_lookup_table.txt\ngs://gatk-sv-resources-public/hg38/v0/sv-resources/ref-panel/1KG/v2/mingq/1KGP_2504_and_698_with_GIAB.1perc_fdr.PCRMINUS.minGQ.filter_lookup_table.txt\ngs://gatk-sv-resources-public/hg38/v0/sv-resources/ref-panel/1KG/v2/mingq/1KGP_2504_and_698_with_GIAB.5perc_fdr.PCRMINUS.minGQ.filter_lookup_table.txt\n"})}),"\n"]}),"\n",(0,n.jsxs)(t.li,{children:["\n",(0,n.jsx)(t.p,{children:"BatchEffect - remove variants that show significant discrepancies\nin allele frequencies across batches"}),"\n"]}),"\n",(0,n.jsxs)(t.li,{children:["\n",(0,n.jsx)(t.p,{children:"FilterOutlierSamplesPostMinGQ - remove outlier samples with unusually\nhigh or low number of SVs"}),"\n"]}),"\n",(0,n.jsxs)(t.li,{children:["\n",(0,n.jsx)(t.p,{children:"FilterCleanupQualRecalibration - sanitize filter columns and\nrecalibrate variant QUAL scores for easier interpretation"}),"\n"]}),"\n"]})]})}function u(e={}){const{wrapper:t}={...(0,r.R)(),...e.components};return t?(0,n.jsx)(t,{...e,children:(0,n.jsx)(d,{...e})}):d(e)}},8453:(e,t,s)=>{s.d(t,{R:()=>o,x:()=>l});var n=s(6540);const r={},i=n.createContext(r);function o(e){const t=n.useContext(i);return n.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function l(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(r):e.components||r:o(e.components),n.createElement(i.Provider,{value:t},e.children)}}}]);