"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[791],{3905:function(e,t,r){r.d(t,{Zo:function(){return p},kt:function(){return m}});var n=r(7294);function l(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}function i(e,t){var r=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),r.push.apply(r,n)}return r}function a(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{};t%2?i(Object(r),!0).forEach((function(t){l(e,t,r[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(r)):i(Object(r)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(r,t))}))}return e}function u(e,t){if(null==e)return{};var r,n,l=function(e,t){if(null==e)return{};var r,n,l={},i=Object.keys(e);for(n=0;n<i.length;n++)r=i[n],t.indexOf(r)>=0||(l[r]=e[r]);return l}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)r=i[n],t.indexOf(r)>=0||Object.prototype.propertyIsEnumerable.call(e,r)&&(l[r]=e[r])}return l}var o=n.createContext({}),s=function(e){var t=n.useContext(o),r=t;return e&&(r="function"==typeof e?e(t):a(a({},t),e)),r},p=function(e){var t=s(e.components);return n.createElement(o.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},c=n.forwardRef((function(e,t){var r=e.components,l=e.mdxType,i=e.originalType,o=e.parentName,p=u(e,["components","mdxType","originalType","parentName"]),c=s(r),m=l,f=c["".concat(o,".").concat(m)]||c[m]||d[m]||i;return r?n.createElement(f,a(a({ref:t},p),{},{components:r})):n.createElement(f,a({ref:t},p))}));function m(e,t){var r=arguments,l=t&&t.mdxType;if("string"==typeof e||l){var i=r.length,a=new Array(i);a[0]=c;var u={};for(var o in t)hasOwnProperty.call(t,o)&&(u[o]=t[o]);u.originalType=e,u.mdxType="string"==typeof e?e:l,a[1]=u;for(var s=2;s<i;s++)a[s]=r[s];return n.createElement.apply(null,a)}return n.createElement.apply(null,r)}c.displayName="MDXCreateElement"},1533:function(e,t,r){r.r(t),r.d(t,{frontMatter:function(){return u},contentTitle:function(){return o},metadata:function(){return s},toc:function(){return p},default:function(){return c}});var n=r(2122),l=r(9756),i=(r(7294),r(3905)),a=["components"],u={sidebar_label:"Module 04",sidebar_position:7},o="Module 04: Merge Cohort VCFs",s={unversionedId:"modules/04",id:"modules/04",isDocsHomePage:!1,title:"Module 04: Merge Cohort VCFs",description:"Combines filtered variants across batches.",source:"@site/docs/modules/04.mdx",sourceDirName:"modules",slug:"/modules/04",permalink:"/gatk-sv/docs/modules/04",editUrl:"https://github.com/broadinstitute/gatk-sv/docs/modules/04.mdx",version:"current",sidebarPosition:7,frontMatter:{sidebar_label:"Module 04",sidebar_position:7},sidebar:"tutorialSidebar",previous:{title:"Module 03",permalink:"/gatk-sv/docs/modules/03"},next:{title:"Module 05/06",permalink:"/gatk-sv/docs/modules/0506"}},p=[{value:"Prerequisites:",id:"prerequisites",children:[]},{value:"Inputs:",id:"inputs",children:[]},{value:"Outputs:",id:"outputs",children:[]},{value:"Module 04",id:"module-04",children:[{value:"Prerequisites:",id:"prerequisites-1",children:[]},{value:"Inputs:",id:"inputs-1",children:[]},{value:"Outputs:",id:"outputs-1",children:[]}]},{value:"Module 04b",id:"module-04b",children:[{value:"Prerequisites:",id:"prerequisites-2",children:[]},{value:"Inputs:",id:"inputs-2",children:[]},{value:"Outputs:",id:"outputs-2",children:[]}]}],d={toc:p};function c(e){var t=e.components,r=(0,l.Z)(e,a);return(0,i.kt)("wrapper",(0,n.Z)({},d,r,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"module-04-merge-cohort-vcfs"},"Module 04: Merge Cohort VCFs"),(0,i.kt)("p",null,"Combines filtered variants across batches.\nThe WDL can be found at: ",(0,i.kt)("inlineCode",{parentName:"p"},"/wdl/MergeCohortVcfs.wdl"),"."),(0,i.kt)("h2",{id:"prerequisites"},"Prerequisites:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"#module03"},"Module 03"))),(0,i.kt)("h2",{id:"inputs"},"Inputs:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"List of filtered PESR VCFs (",(0,i.kt)("a",{parentName:"li",href:"03"},"Module 03"),")"),(0,i.kt)("li",{parentName:"ul"},"List of filtered depth VCFs (",(0,i.kt)("a",{parentName:"li",href:"03"},"Module 03"),")")),(0,i.kt)("h2",{id:"outputs"},"Outputs:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Combined cohort PESR and depth VCFs"),(0,i.kt)("li",{parentName:"ul"},"Cohort and clustered depth variant BED files")),(0,i.kt)("h2",{id:"module-04"},"Module 04"),(0,i.kt)("p",null,"Genotypes a batch of samples across unfiltered\nvariants combined across all batches."),(0,i.kt)("h3",{id:"prerequisites-1"},"Prerequisites:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"03"},"Module 03")),(0,i.kt)("li",{parentName:"ul"},"Merge Cohort VCFs")),(0,i.kt)("h3",{id:"inputs-1"},"Inputs:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Batch PESR and depth VCFs (",(0,i.kt)("a",{parentName:"li",href:"03"},"Module 03"),")"),(0,i.kt)("li",{parentName:"ul"},"Cohort PESR and depth VCFs (Merge Cohort VCFs)"),(0,i.kt)("li",{parentName:"ul"},"Batch read count, PE, and SR files (",(0,i.kt)("a",{parentName:"li",href:"00c"},"Module 00c"),")")),(0,i.kt)("h3",{id:"outputs-1"},"Outputs:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},'Filtered SV (non-depth-only a.k.a. "PESR") VCF\nwith outlier samples excluded'),(0,i.kt)("li",{parentName:"ul"},"Filtered depth-only call VCF with outlier samples excluded"),(0,i.kt)("li",{parentName:"ul"},"PED file with outlier samples excluded"),(0,i.kt)("li",{parentName:"ul"},"List of SR pass variants"),(0,i.kt)("li",{parentName:"ul"},"List of SR fail variants"),(0,i.kt)("li",{parentName:"ul"},"(Optional) Depth re-genotyping intervals list")),(0,i.kt)("h2",{id:"module-04b"},"Module 04b"),(0,i.kt)("p",null,"Re-genotypes probable mosaic variants across multiple batches."),(0,i.kt)("h3",{id:"prerequisites-2"},"Prerequisites:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"04"},"Module 04"))),(0,i.kt)("h3",{id:"inputs-2"},"Inputs:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Per-sample median coverage estimates (",(0,i.kt)("a",{parentName:"li",href:"00c"},"Module 00c"),")"),(0,i.kt)("li",{parentName:"ul"},"Pre-genotyping depth VCFs (",(0,i.kt)("a",{parentName:"li",href:"03"},"Module 03"),")"),(0,i.kt)("li",{parentName:"ul"},"Batch PED files (",(0,i.kt)("a",{parentName:"li",href:"03"},"Module 03"),")"),(0,i.kt)("li",{parentName:"ul"},"Clustered depth variant BED file (Merge Cohort VCFs)"),(0,i.kt)("li",{parentName:"ul"},"Cohort depth VCF (Merge Cohort VCFs)"),(0,i.kt)("li",{parentName:"ul"},"Genotyped depth VCFs (",(0,i.kt)("a",{parentName:"li",href:"04"},"Module 04"),")"),(0,i.kt)("li",{parentName:"ul"},"Genotyped depth RD cutoffs file (",(0,i.kt)("a",{parentName:"li",href:"04"},"Module 04"),")")),(0,i.kt)("h3",{id:"outputs-2"},"Outputs:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Re-genotyped depth VCFs")))}c.isMDXComponent=!0}}]);