"use strict";(self.webpackChunkGATK_SV=self.webpackChunkGATK_SV||[]).push([[3416],{4295:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>d,contentTitle:()=>l,default:()=>p,frontMatter:()=>s,metadata:()=>a,toc:()=>c});var i=t(4848),o=t(8453),r=t(1944);const s={title:"TrainGCNV",description:"Train gCNV",sidebar_position:3,slug:"gcnv"},l=void 0,a={id:"modules/train_gcnv",title:"TrainGCNV",description:"Train gCNV",source:"@site/docs/modules/train_gcnv.md",sourceDirName:"modules",slug:"/modules/gcnv",permalink:"/gatk-sv/docs/modules/gcnv",draft:!1,unlisted:!1,editUrl:"https://github.com/broadinstitute/gatk-sv/tree/master/website/docs/modules/train_gcnv.md",tags:[],version:"current",sidebarPosition:3,frontMatter:{title:"TrainGCNV",description:"Train gCNV",sidebar_position:3,slug:"gcnv"},sidebar:"tutorialSidebar",previous:{title:"EvidenceQC",permalink:"/gatk-sv/docs/modules/eqc"},next:{title:"GatherBatchEvidence",permalink:"/gatk-sv/docs/modules/gbe"}},d={},c=[{value:"Inputs",id:"inputs",level:2},{value:"<code>samples</code>",id:"samples",level:4},{value:"<code>count_files</code>",id:"count_files",level:4},{value:"<code>contig_ploidy_priors</code>",id:"contig_ploidy_priors",level:4},{value:"<code>reference_fasta</code>",id:"reference_fasta",level:4},{value:"Outputs",id:"outputs",level:2},{value:"<HighlightOptionalArg>Optional</HighlightOptionalArg> <code>annotated_intervals</code>",id:"annotated-intervals",level:4},{value:"<HighlightOptionalArg>Optional</HighlightOptionalArg> <code>filtered_intervals_cnv</code>",id:"filtered-intervals-cnv",level:4},{value:"<HighlightOptionalArg>Optional</HighlightOptionalArg> <code>cohort_contig_ploidy_model_tar</code>",id:"contig-ploidy-model-tarball",level:4},{value:"<HighlightOptionalArg>Optional</HighlightOptionalArg> <code>cohort_gcnv_model_tars</code>",id:"model-tarballs",level:4}];function h(e){const n={a:"a",code:"code",em:"em",h2:"h2",h4:"h4",mermaid:"mermaid",p:"p",...(0,o.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsxs)(n.p,{children:[(0,i.jsx)(n.a,{href:"https://www.nature.com/articles/s41588-023-01449-0",children:"GATK-gCNV"}),"\nis a method for detecting rare germline copy number variants (CNVs)\nfrom short-read sequencing read-depth information.\nThe ",(0,i.jsx)(n.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/wdl/TrainGCNV.wdl",children:"TrainGCNV"}),"\nmodule trains a gCNV model for use in the ",(0,i.jsx)(n.a,{href:"./gbe",children:"GatherBatchEvidence"})," workflow.\nThe upstream and downstream dependencies of the TrainGCNV module are illustrated in the following diagram."]}),"\n",(0,i.jsx)(n.p,{children:"The samples used for training should be homogeneous (concerning sequencing platform,\ncoverage, library preparation, etc.) and similar\nto the samples on which the model will be applied in terms of sample type,\nlibrary preparation protocol, sequencer, sequencing center, and etc."}),"\n",(0,i.jsxs)(n.p,{children:["For small, relatively homogeneous cohorts, a single gCNV model is usually sufficient.\nHowever, for larger cohorts, especially those with multiple data sources,\nwe recommend training a separate model for each batch or group of batches (see\n",(0,i.jsx)(n.a,{href:"/docs/run/joint#batching",children:"batching section"})," for details).\nThe model can be trained on all or a subset of the samples to which it will be applied.\nA subset of 100 randomly selected samples from the batch is a reasonable\ninput size for training the model; when the ",(0,i.jsx)(n.code,{children:"n_samples_subsample"})," input is provided,\nthe ",(0,i.jsx)(n.code,{children:"TrainGCNV"})," workflow can automatically perform this random selection."]}),"\n",(0,i.jsxs)(n.p,{children:["The following diagram illustrates the upstream and downstream workflows of the ",(0,i.jsx)(n.code,{children:"TrainGCNV"})," workflow\nin the recommended invocation order. You may refer to\n",(0,i.jsx)(n.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/terra_pipeline_diagram.jpg",children:"this diagram"}),"\nfor the overall recommended invocation order."]}),"\n",(0,i.jsx)(n.mermaid,{value:"\nstateDiagram\n  direction LR\n  \n  classDef inModules stroke-width:0px,fill:#00509d,color:#caf0f8\n  classDef thisModule font-weight:bold,stroke-width:0px,fill:#ff9900,color:white\n  classDef outModules stroke-width:0px,fill:#caf0f8,color:#00509d\n\n  batching: Batching, sample QC, and sex assignment\n  t: TrainGCNV\n  gbe: GatherBatchEvidence\n  \n  batching --\x3e t\n  t --\x3e gbe \n  \n  class t thisModule\n  class batching inModules\n  class gbe outModules"}),"\n",(0,i.jsx)(n.h2,{id:"inputs",children:"Inputs"}),"\n",(0,i.jsxs)(n.p,{children:["This section provides a brief description on the ",(0,i.jsx)(n.em,{children:"required"})," inputs of the TrainGCNV workflow.\nFor a description on the ",(0,i.jsx)(n.em,{children:"optional"})," inputs and their default values, you may refer to the\n",(0,i.jsx)(n.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/wdl/TrainGCNV.wdl",children:"source code"})," of the TrainGCNV workflow.\nAdditionally, the majority of the optional inputs of the workflow map to the optional arguments of the\ntool the workflow uses, ",(0,i.jsx)(n.code,{children:"GATK GermlineCNVCaller"}),"; hence, you may refer to the\n",(0,i.jsx)(n.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360040097712-GermlineCNVCaller",children:"documentation"}),"\nof the tool for a description on these optional inputs."]}),"\n",(0,i.jsx)(n.h4,{id:"samples",children:(0,i.jsx)(n.code,{children:"samples"})}),"\n",(0,i.jsxs)(n.p,{children:["A list of sample IDs.\nThe order of IDs in this list should match the order of files in ",(0,i.jsx)(n.code,{children:"count_files"}),"."]}),"\n",(0,i.jsx)(n.h4,{id:"count_files",children:(0,i.jsx)(n.code,{children:"count_files"})}),"\n",(0,i.jsxs)(n.p,{children:["A list of per-sample coverage counts generated in the ",(0,i.jsx)(n.a,{href:"./gse#outputs",children:"GatherSampleEvidence"})," workflow."]}),"\n",(0,i.jsx)(n.h4,{id:"contig_ploidy_priors",children:(0,i.jsx)(n.code,{children:"contig_ploidy_priors"})}),"\n",(0,i.jsxs)(n.p,{children:["A tabular file with ploidy prior probability per contig.\nYou may find the link to this input from\n",(0,i.jsx)(n.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/values/resources_hg38.json",children:"this reference"}),"\nand a description to the file format\n",(0,i.jsx)(n.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360037224772-DetermineGermlineContigPloidy",children:"here"}),"."]}),"\n",(0,i.jsx)(n.h4,{id:"reference_fasta",children:(0,i.jsx)(n.code,{children:"reference_fasta"})}),"\n",(0,i.jsxs)(n.p,{children:[(0,i.jsx)(n.code,{children:"reference_fasta"}),", ",(0,i.jsx)(n.code,{children:"reference_index"}),", ",(0,i.jsx)(n.code,{children:"reference_dict"})," are respectively the\nreference genome sequence in the FASTA format, its index file, and a corresponding\n",(0,i.jsx)(n.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035531652-FASTA-Reference-genome-format",children:"dictionary file"}),".\nYou may find links to these files from\n",(0,i.jsx)(n.a,{href:"https://github.com/broadinstitute/gatk-sv/blob/main/inputs/values/resources_hg38.json",children:"this reference"}),"."]}),"\n",(0,i.jsx)(n.h2,{id:"outputs",children:"Outputs"}),"\n",(0,i.jsxs)(n.h4,{id:"annotated-intervals",children:[(0,i.jsx)(r.$,{children:"Optional"})," ",(0,i.jsx)(n.code,{children:"annotated_intervals"})]}),"\n",(0,i.jsxs)(n.p,{children:["The count files from ",(0,i.jsx)(n.a,{href:"./gse",children:"GatherSampleEvidence"})," with adjacent intervals combined into\nlocus-sorted ",(0,i.jsx)(n.code,{children:"DepthEvidence"})," files using ",(0,i.jsx)(n.code,{children:"GATK CondenseDepthEvidence"})," tool, which are\nannotated with GC content, mappability, and segmental-duplication content using\n",(0,i.jsx)(n.a,{href:"https://gatk.broadinstitute.org/hc/en-us/articles/360041416652-AnnotateIntervals",children:(0,i.jsx)(n.code,{children:"GATK AnnotateIntervals"})}),"\ntool. This output is generated if the optional input ",(0,i.jsx)(n.code,{children:"do_explicit_gc_correction"})," is set to ",(0,i.jsx)(n.code,{children:"True"}),"."]}),"\n",(0,i.jsxs)(n.h4,{id:"filtered-intervals-cnv",children:[(0,i.jsx)(r.$,{children:"Optional"})," ",(0,i.jsx)(n.code,{children:"filtered_intervals_cnv"})]}),"\n",(0,i.jsxs)(n.h4,{id:"contig-ploidy-model-tarball",children:[(0,i.jsx)(r.$,{children:"Optional"})," ",(0,i.jsx)(n.code,{children:"cohort_contig_ploidy_model_tar"})]}),"\n",(0,i.jsxs)(n.h4,{id:"model-tarballs",children:[(0,i.jsx)(r.$,{children:"Optional"})," ",(0,i.jsx)(n.code,{children:"cohort_gcnv_model_tars"})]})]})}function p(e={}){const{wrapper:n}={...(0,o.R)(),...e.components};return n?(0,i.jsx)(n,{...e,children:(0,i.jsx)(h,{...e})}):h(e)}},1944:(e,n,t)=>{t.d(n,{$:()=>o});var i=t(4848);const o=e=>{let{children:n}=e;return(0,i.jsx)("span",{style:{backgroundColor:"var(--highlight-optional-arg-background-color)",borderRadius:"2px",color:"var(--highlight-optional-arg-text-color)",padding:"0.2rem"},children:n})}},8453:(e,n,t)=>{t.d(n,{R:()=>s,x:()=>l});var i=t(6540);const o={},r=i.createContext(o);function s(e){const n=i.useContext(r);return i.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function l(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(o):e.components||o:s(e.components),i.createElement(r.Provider,{value:n},e.children)}}}]);