---
sidebar_label: Requirements
sidebar_position: 3
---

# Requirements


## Deployment and execution:
* A [Google Cloud](https://cloud.google.com/) account.
* A workflow execution system supporting the
[Workflow Description Language](https://openwdl.org/) (WDL), either:
  * [Cromwell](https://github.com/broadinstitute/cromwell)
  (v36 or higher). A dedicated server is highly recommended.
  * or [Terra](https://terra.bio/) (note preconfigured
  GATK-SV workflows are not yet available for this platform)
* Recommended: [MELT](https://melt.igs.umaryland.edu/).
Due to licensing restrictions, we cannot provide a public
docker image or reference panel VCFs for this algorithm.
* Recommended: [cromshell](https://github.com/broadinstitute/cromshell)
for interacting with a dedicated Cromwell server.
* Recommended: [WOMtool](https://cromwell.readthedocs.io/en/stable/WOMtool/)
for validating WDL/json files.

## Data:
* Illumina short-read whole-genome CRAMs or BAMs, aligned
to hg38 with [bwa-mem](https://github.com/lh3/bwa).
BAMs must also be indexed.
* Indexed GVCFs produced by GATK HaplotypeCaller, or a jointly genotyped VCF.
* Family structure definitions file in
[PED format](https://gatk.broadinstitute.org/hc/en-us/articles/360035531972-PED-Pedigree-format).
Sex aneuploidies (detected in [Module 00b](../modules/00b))
should be entered as sex = 0.

## Sample ID requirements

Sample IDs must:
* Be unique within the cohort
* Contain only alphanumeric characters and underscores
(no dashes, whitespace, or special characters)

Sample IDs should not:
* Contain only numeric characters
* Be a substring of another sample ID in the same cohort
* Contain any of the following substrings: `chr`, `name`,
`DEL`, `DUP`, `CPX`, `CHROM`

The same requirements apply to family IDs in the PED file,
as well as batch IDs and the cohort ID provided as workflow inputs.

Sample IDs are provided to [Module00a](../modules/00a) directly
and need not match sample names from the BAM/CRAM headers or
GVCFs. `GetSampleID.wdl` can be used to fetch BAM sample IDs
and also generates a set of alternate IDs that are considered
safe for this pipeline; alternatively,
[this script](https://github.com/talkowski-lab/gnomad_sv_v3/blob/master/sample_id/convert_sample_ids.py)
transforms a list of sample IDs to fit these requirements.
Currently, sample IDs can be replaced again in [Module 00c](../modules/00c).

The following inputs will need to be updated with the transformed sample IDs:
* Sample ID list for [Module00a](../modules/00a) or [Module 00c](../modules/00c)
* PED file

If using a SNP VCF in [Module 00c](../modules/00c), it does not
need to be re-headered; simply provide the `vcf_samples` argument.
