---
sidebar_position: 3
---

# Module 00c

Runs CNV callers (cnMOPs, GATK gCNV) and combines
single-sample raw evidence into a batch.
See [cohort-mode](../getting_started/overview#cohort-mode)
for more information on batching.

## Prerequisites:
* [Module 00a](00a)
* (Recommended) [Module 00b](00b)
* gCNV training

## Inputs:
* PED file (updated with [Module 00b](00b) sex
assignments, including sex = 0 for sex
aneuploidies. Calls will not be made on sex
chromosomes when sex = 0 in order to avoid
generating many confusing calls or upsetting
normalized copy numbers for the batch.)
* Per-sample GVCFs generated with HaplotypeCaller
(`gvcfs` input), or a jointly-genotyped VCF
(position-sharded, `snp_vcfs` input or
`snp_vcfs_shard_list` input)
* Read count, BAF, PE, and SR files ([Module 00a](00a))
* Caller VCFs ([Module 00a](00a))
* Contig ploidy model and gCNV model files (gCNV training)

## Outputs:
* Combined read count matrix, SR, PE, and BAF files
* Standardized call VCFs
* Depth-only (DEL/DUP) calls
* Per-sample median coverage estimates
* (Optional) Evidence QC plots


